lib/pos_epoch_complete_hook.go

```diff
@@ -138,12 +138,28 @@ func (bav *UtxoView) runEpochCompleteSnapshotGeneration(epochNumber uint64) erro
 	return nil
 }
 
+// simulateNextEpochEntry simulates the block range for the next epoch given the current epoch's final
+// block height and epoch number. The view and timestamp for the simulated epoch are left empty since they can't
+// be easily simulated, so DO NOT USE CreatedAtBlockTimestampNanoSecs or InitialView from the returned EpochEntry.
+func (bav *UtxoView) simulateNextEpochEntry(currentEpochNumber uint64, currentEpochFinalBlockHeight uint64) (*EpochEntry, error) {
+	return bav.computeNextEpochEntry(
+		currentEpochNumber,
+		currentEpochFinalBlockHeight,
+		0,
+		0,
+	)
+}
+
 func (bav *UtxoView) computeNextEpochEntry(currentEpochNumber uint64, currentEpochFinalBlockHeight uint64, currentEpochFinalView uint64, nextEpochBlockTimestampNanoSecs uint64) (*EpochEntry, error) {
 	// Retrieve the SnapshotGlobalParamsEntry to determine the next epoch's final block height. We use the
 	// snapshot global params here because the next epoch begin immediately, and its length is used in the PoS
 	// consensus. The validator set for the next epoch needs to be in agreement on the length of the epoch
 	// before the epoch begins.
-	snapshotGlobalParamsEntry, err := bav.GetCurrentSnapshotGlobalParamsEntry()
+	snapshotAtEpochNumber, err := bav.ComputeSnapshotEpochNumberForEpoch(currentEpochNumber)
```

Comment:
We shouldn't assume that the `currentEpochNumber` value passed in is the utxoView's current epoch value